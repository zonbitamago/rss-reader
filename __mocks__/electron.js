const remote ={
  app : {
    getPath :function(str){return '.'}
  }
}

module.exports = {
  require: jest.genMockFunction(),
  match: jest.genMockFunction(),
  app: jest.genMockFunction(),
  remote: remote,
  dialog: jest.genMockFunction()
};
