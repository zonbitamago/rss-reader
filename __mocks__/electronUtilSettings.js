export const electronUtil = {
  closeApp: () => {
    console.log("closeApp");
  },
  minimizeApp: () => {
    console.log("minimizeApp");
  },
  openBrowser: url => {
    console.log("openBrowser");
    console.log(url);
  }
};
