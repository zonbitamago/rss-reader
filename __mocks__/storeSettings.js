import { resolve } from "bluebird-lst";
import moment from "moment";

export let store = {
  ItemStore: {
    items: [],
    loading: false,
    updateDate: moment("2012-06-20 09:12:34").format("HH:mm:ss"),
    updateDuration: 5,
    timerId: "dummy",
    add: () => console.log("add"),
    getSettings: () => console.log("getSettings"),
    setSetting: () => console.log("setSetting"),
    setTimer: () => console.log("setTimer")
  },
  FeedListStore: {
    feedList: [
      { name: "google", url: "https://www.google.co.jp/" },
      { name: "yahoo", url: "https://www.yahoo.co.jp/" }
    ],
    getFeedList: () => console.log("getFeedList"),
    setFeedList: () => {
      console.log("setFeedList");
      return new Promise(resolve).then(() => {
        return true;
      });
    },
    deleteFeedList: () => console.log("deleteFeedList")
  }
};
