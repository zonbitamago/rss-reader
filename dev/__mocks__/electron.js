module.exports = {
  ipcRenderer: {
    send: jest.fn()
  },
  remote: {
    app: {
      getPath: () => {
        return "";
      }
    }
  }
};
