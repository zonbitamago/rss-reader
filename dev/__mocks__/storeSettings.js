export let store = {
  ItemStore: { items: [], add: () => console.log("add") },
  SettingStore: {
    updateDuration: 5,
    getSettings: () => console.log("getSettings"),
    setSetting: () => console.log("setSetting")
  },
  RssListStore: {
    rssList: [
      { name: "google", url: "https://www.google.co.jp/" },
      { name: "yahoo", url: "https://www.yahoo.co.jp/" }
    ],
    getRssList: () => console.log("getRssList"),
    setRssList: () => console.log("setRssList"),
    deleteRssList: () => console.log("deleteRssList")
  }
};
