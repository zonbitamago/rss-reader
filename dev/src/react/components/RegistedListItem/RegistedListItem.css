.RegistedListItem {
}

.RegistedListItem-Text > h3 {
  color: #42a5f5;
  text-decoration: underline;
}
