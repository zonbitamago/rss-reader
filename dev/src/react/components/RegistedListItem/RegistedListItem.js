import React, { Component } from "react";
import PropTypes from "prop-types";
import styles from "./RegistedListItem.css";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import IconButton from "@material-ui/core/IconButton";
import Delete from "@material-ui/icons/Delete";

class RegistedListItem extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="RegistedListItem">
        <ListItem component="a" href={this.props.url}>
          <ListItemText
            className="RegistedListItem-Text"
            primary={this.props.text}
          />
          <ListItemSecondaryAction>
            <IconButton aria-label="Delete">
              <Delete />
            </IconButton>
          </ListItemSecondaryAction>
        </ListItem>
      </div>
    );
  }
}

RegistedListItem.propTypes = {
  text: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};

RegistedListItem.defaultProps = {};

export default RegistedListItem;
