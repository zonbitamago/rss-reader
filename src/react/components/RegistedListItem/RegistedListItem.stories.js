import React from "react";
import { storiesOf } from "@storybook/react";
import RegistedListItem from "./RegistedListItem";
import List from "@material-ui/core/List";
import { electronUtil } from "../../../../__mocks__/electronUtilSettings";
import { store } from "../../../../__mocks__/storeSettings";

storiesOf("Atoms/RegistedListItem", module)
  .addDecorator(story => <List>{story()}</List>)
  .add("RegistedListItem", () => (
    <RegistedListItem
      name="sample"
      url="https://www.google.co.jp/"
      store={store}
      electronUtil={electronUtil}
    />
  ));
