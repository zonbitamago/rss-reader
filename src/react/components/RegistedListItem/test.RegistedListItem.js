import React from "react";
import { shallow, mount } from "enzyme";

import RegistedListItem from "./RegistedListItem";
import { electronUtil } from "../../../../__mocks__/electronUtilSettings";
import { store } from "../../../../__mocks__/storeSettings";

jest.mock("../../../../__mocks__/electronUtilSettings");
jest.mock("../../../../__mocks__/storeSettings");

describe("RegistedListItem", () => {
  let component, wrapper, props;

  beforeEach(() => {
    electronUtil.openBrowser.mockRestore();
    props = {
      name: "test",
      url: "https://google/co.jp",
      electronUtil: electronUtil,
      store: store
    };
  });

  it("should", () => {
    component = shallow(<RegistedListItem {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("link click", () => {
    expect(electronUtil.openBrowser.mock.calls.length).toBe(0);
    wrapper = mount(<RegistedListItem {...props} />);

    wrapper
      .find(".RegistedListItem .RegistedListItem-Text div")
      .simulate("click");

    expect(electronUtil.openBrowser.mock.calls.length).toBe(1);
  });

  it("Delete Icon Click", () => {
    expect(store.FeedListStore.deleteFeedList.mock.calls.length).toBe(0);
    expect(store.ItemStore.setTimer.mock.calls.length).toBe(0);
    wrapper = mount(<RegistedListItem {...props} />);

    wrapper.find(".RegistedListItem svg").simulate("click");

    expect(store.FeedListStore.deleteFeedList.mock.calls.length).toBe(1);
    expect(store.ItemStore.setTimer.mock.calls.length).toBe(1);
  });
});
