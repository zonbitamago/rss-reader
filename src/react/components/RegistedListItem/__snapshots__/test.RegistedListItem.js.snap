// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistedListItem should 1`] = `
<div
  className="RegistedListItem"
>
  <WithStyles(ListItem)
    component="a"
  >
    <WithStyles(ListItemText)
      className="RegistedListItem-Text"
      onClick={[Function]}
      primary="test"
    />
    <WithStyles(ListItemSecondaryAction)>
      <WithStyles(IconButton)
        aria-label="Delete"
      >
        <pure(Delete)
          onClick={[Function]}
        />
      </WithStyles(IconButton)>
    </WithStyles(ListItemSecondaryAction)>
  </WithStyles(ListItem)>
</div>
`;
